### ISSUE TITLE
x64dbg debug not working

### ISSUE THREAD

**Issue by abc123098123** [sonic2kk/steamtinkerlaunch#947]
## System Information
- SteamTinkerLaunch version: v14.0.20231021-2
- Distribution: Arch Linux
- Installation Method: Package Manager

## Issue Description
I select the option to `Debug with x64dbg` and put the game in a steam collection called `x64debug` but the game always starts without x64dbg.

From the logs, that you can find below, it seems `steamtinkerlaunch` tries to download x64dbg without success and because of that, x64dbg archive extraction does not work.

I also tried to install x64dbg in the expected directory. After that, it does recognize that x64dbg exists and tries to run it, giving the following error, extracted from other logs:

```
Sat Oct 21 03:40:09 PM WEST 2023 INFO - extWine64Run - Command in Proton WINEPREFIX is: WINE="" WINEARCH=win64 WINEDEBUG="-all" WINEPREFIX="/run/media/romeu/GAMES/SteamLibrary/steamapps/compatdata/230050/pfx"  /home/romeu/.config/steamtinkerlaunch/downloads/x64dbg/release/x32/x32dbg.exe /usr/bin/steamtinkerlaunch: line 6872: : command not found
```

It says `command not found`, but I don't know which command. Steamtinkerlaunch is inside `/usr/bin/steamtinkerlaunch`, and `x32dbg` executable file is inside the expected directory

## Logs

```
Sat Oct 21 03:33:06 PM WEST 2023 INFO - setflatpak - started SteamTinkerLaunch from /usr/bin/steamtinkerlaunch
Sat Oct 21 03:33:06 PM WEST 2023 INFO - initAID - Set AID to PLACEHOLDERAID '31337'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - main - Current SteamTinkerLaunch working directory is '/home/romeu'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLanguage - First load the default language 'english' to make sure all variables are filled
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - Language from command line is 'english'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - SYSTEMSTLCFGDIR is '/usr/share/steamtinkerlaunch'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - Command line language 'english' is no file - trying to find its absolute path
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - Loading found system wide /usr/share/steamtinkerlaunch/lang//english.txt
Sat Oct 21 03:33:06 PM WEST 2023 WARN - updateEditor - 'xdg-open' selected as editor or configured editor not found - trying to find an installed editor installed
Sat Oct 21 03:33:06 PM WEST 2023 INFO - updateEditor - No valid editor found - will fall back to 'xdg-open'.
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLanguage - Loading STLLANG from '/home/romeu/.config/steamtinkerlaunch/global.conf'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLanguage - STLLANG from '/home/romeu/.config/steamtinkerlaunch/global.conf' is 'english'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - steamdedeckt - Not on Steam Deck I guess
Sat Oct 21 03:33:06 PM WEST 2023 INFO - setLocalInstall - Looks like we don't have a local non-root install
Sat Oct 21 03:33:06 PM WEST 2023 INFO - updateConfigEntry - Option 'GLOBALCOLLECTIONDIR' is updated with the new value '/usr/share/steamtinkerlaunch/collections' in config 'global.conf'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - getCurrentCommandline - No Proton in command line arguments '--version'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - saveOrgVars - Storing some original variables to restore them later
Sat Oct 21 03:33:06 PM WEST 2023 INFO - emptyVars - Emptied some original variables as they slowdown several system calls when started from steam
Sat Oct 21 03:33:06 PM WEST 2023 INFO - emptyVars - Set $PATH to '/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/android-ndk:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin:/home/romeu/.local/bin:/home/romeu/.local/bin/fzf:/home/romeu/.local/bin/ffmpeg:/home/romeu/.local/share/cargo/bin:/home/romeu/.cache/fvm/default/bin:/home/romeu/.cargo/bin:/home/romeu/.gem/ruby/2.5.0/bin:/home/romeu/.gem/ruby/2.6.0/bin:/home/romeu/.gem/ruby/2.7.0/bin:/home/romeu/.gem/ruby/3.0.0/bin:/home/romeu/.local/share/gem/ruby/3.0.0/bin:/home/romeu/Android/Sdk/build-tools/33.0.1:/home/romeu/.fzf/bin:'
Sat Oct 21 03:33:06 PM WEST 2023 START - ######### Initializing Game Launch 31337 using SteamTinkerLaunch v14.0.20231021-2 #########
Sat Oct 21 03:33:06 PM WEST 2023 INFO - main - Starting with a clean log
Sat Oct 21 03:33:06 PM WEST 2023 INFO - main - Start creating default configs
Sat Oct 21 03:33:06 PM WEST 2023 INFO - createDefaultCfgs - START
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLanguage - First load the default language 'english' to make sure all variables are filled
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - Language from command line is 'english'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - GLOBALSTLLANGDIR is '/usr/share/steamtinkerlaunch/lang'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - Command line language 'english' is no file - trying to find its absolute path
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLangFile - Loading found system wide /usr/share/steamtinkerlaunch/lang/english.txt
Sat Oct 21 03:33:06 PM WEST 2023 WARN - updateEditor - 'xdg-open' selected as editor or configured editor not found - trying to find an installed editor installed
Sat Oct 21 03:33:06 PM WEST 2023 INFO - updateEditor - No valid editor found - will fall back to 'xdg-open'.
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLanguage - Loading STLLANG from '/home/romeu/.config/steamtinkerlaunch/global.conf'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - loadLanguage - STLLANG from '/home/romeu/.config/steamtinkerlaunch/global.conf' is 'english'
Sat Oct 21 03:33:06 PM WEST 2023 INFO - createDefaultCfgs - setSteamPaths:
Sat Oct 21 03:33:06 PM WEST 2023 INFO - setSteamPaths - Reading Steam Path variables from '/dev/shm/steamtinkerlaunch/steampaths.txt'
Sat Oct 21 03:33:06 PM WEST 2023 WARN - updateEditor - 'xdg-open' selected as editor or configured editor not found - trying to find an installed editor installed
Sat Oct 21 03:33:06 PM WEST 2023 INFO - updateEditor - No valid editor found - will fall back to 'xdg-open'.
Sat Oct 21 03:33:06 PM WEST 2023 INFO - getGameOS - Starting game OS detection
Sat Oct 21 03:33:06 PM WEST 2023 INFO - getGameOS - STEAM_COMPAT_DATA_PATH is not defined, so this is either a Linux Game or no game was started at all
Sat Oct 21 03:33:07 PM WEST 2023 INFO - listAllSettingsEntries - '/dev/shm/steamtinkerlaunch/setentries.txt' already exists - nothing to do
Sat Oct 21 03:33:07 PM WEST 2023 INFO - createDefaultCfgs - STOP
Sat Oct 21 03:33:07 PM WEST 2023 SKIP - listSteamLibraries - not recreating already available '/dev/shm/steamtinkerlaunch/SteamLibraries.txt'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - setSteamLibraryPaths - STEAM_COMPAT_LIBRARY_PATHS set to '/home/romeu/.local/share/Steam/steamapps:/run/media/romeu/GAMES/SteamLibrary/steamapps:/run/media/romeu/SSD_ONE/SteamLibrary/steamapps'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - setSteamLibraryPaths - STEAM_COMPAT_MOUNTS set to '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper:/home/romeu/.local/share/Steam/steamapps/common/Steamworks Shared:/run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - setSteamLibraryPaths - STEAM_COMPAT_TOOL_PATHS set to '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper:/home/romeu/.local/share/Steam/steamapps/common/Steamworks Shared:/run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - main - Checking internal dependencies:
Sat Oct 21 03:33:07 PM WEST 2023 INFO - setAwkBin - Found 'gawk' as an 'awk' variant. It should work without any issues, because 'gawk' was tested completely
Sat Oct 21 03:33:07 PM WEST 2023 INFO - checkIntDeps - Result of version check for yad binary '/usr/bin/yad' is '13.0'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - checkIntDeps - Yad is set correctly in the Global Config, nothing to do.
Sat Oct 21 03:33:07 PM WEST 2023 INFO - main - Initializing first Proton:
Sat Oct 21 03:33:07 PM WEST 2023 INFO - initOldProtonArr - Creating an initial array with available Proton versions using the file '/dev/shm/steamtinkerlaunch/ProtonCSV.txt' which was created during a previous run
Sat Oct 21 03:33:07 PM WEST 2023 INFO - initFirstProton - Initializing Proton
Sat Oct 21 03:33:07 PM WEST 2023 INFO - initFirstProton - Initial Proton version 'FIRSTUSEPROTON' from '/home/romeu/.config/steamtinkerlaunch/default_template.conf' is 'proton-8.0-4b'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - main - Initializing default window resolution
Sat Oct 21 03:33:07 PM WEST 2023 INFO - setInitWinXY - Using '2880x1620' from config '/dev/shm/steamtinkerlaunch/defres.txt'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - main - Welcome to SteamTinkerLaunch
Sat Oct 21 03:33:07 PM WEST 2023 INFO - removeEmptyFiles - Removing empty files from '/home/romeu/.config/steamtinkerlaunch/games/appinfo'
Sat Oct 21 03:33:07 PM WEST 2023 INFO - removeEmptyFiles - Removing empty files from '/home/romeu/.config/steamtinkerlaunch/games/header'
Sat Oct 21 03:33:07 PM WEST 2023 SKIP - setGDKBackend - XDG_SESSION_TYPE is defined and is not Wayland, it is 'tty' - No need to set GDK_BACKEND=x11 as it will already default to X11
Sat Oct 21 03:33:07 PM WEST 2023 INFO - main - Checking command line: incoming arguments '--version'
Sat Oct 21 03:34:56 PM WEST 2023 INFO - openMenu - Clicked '6' - 'SAVE'
Sat Oct 21 03:34:56 PM WEST 2023 INFO - openMenu - exiting 'Favorite Menu' and opening 'openGameMenu'
Sat Oct 21 03:34:56 PM WEST 2023 INFO - openMenu - Saving config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' and restart 'openGameMenu' GUI
Sat Oct 21 03:34:56 PM WEST 2023 INFO - saveMenuEntries - Saving changed Settings
Sat Oct 21 03:34:56 PM WEST 2023 INFO - saveMenuEntries - Clearing Results array with '0' elements
Sat Oct 21 03:34:56 PM WEST 2023 INFO - saveMenuEntries - Pulling values from tempfile '/dev/shm/steamtinkerlaunch/menu.UcZZeZZu'
Sat Oct 21 03:34:57 PM WEST 2023 INFO - updateWinRes - The Window 'SteamTinkerLaunch-launchGameMenu' was closed - the resolution didn't change - nothing to do
Sat Oct 21 03:34:57 PM WEST 2023 INFO - updateConfigEntry - Option 'DXVK_ASYNC' is updated with the new value '1' in config '230050.conf'
Sat Oct 21 03:34:58 PM WEST 2023 INFO - saveMenuEntries - Done with Saving changed Settings
Sat Oct 21 03:34:59 PM WEST 2023 INFO - prepareMenu - Opening Menu
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/global.conf' START
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/global.conf' STOP
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' START
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - Loading game config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - migrateCfgOption - File '/home/romeu/.config/steamtinkerlaunch/dxvk/230050.conf' does not exist - nothing to do
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' STOP
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/url.conf' START
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/url.conf' STOP
Sat Oct 21 03:34:59 PM WEST 2023 SKIP - updateConfigFile - Config file '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' already at version 'v14.0.20231021-2'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - linkGameCfg - Symlink '/home/romeu/.config/steamtinkerlaunch/gamecfgs/title/DLC Quest.conf' already exists
Sat Oct 21 03:34:59 PM WEST 2023 INFO - createWineList - Updating the Wine Dropdown List
Sat Oct 21 03:34:59 PM WEST 2023 INFO - getGameTextFiles - Searching for at least one editable textfile in '/run/media/romeu/GAMES/SteamLibrary/steamapps/compatdata/230050/pfx/drive_c/users/steamuser'
Sat Oct 21 03:34:59 PM WEST 2023 SKIP - updateConfigFile - Config file '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' already at version 'v14.0.20231021-2'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - linkGameCfg - Symlink '/home/romeu/.config/steamtinkerlaunch/gamecfgs/title/DLC Quest.conf' already exists
Sat Oct 21 03:34:59 PM WEST 2023 INFO - createWineList - Updating the Wine Dropdown List
Sat Oct 21 03:34:59 PM WEST 2023 INFO - rmCfgTemp - Deleting and resetting MKCFG '/dev/shm/steamtinkerlaunch/menu.UcZZeZZu' - triggered from 'mkCfgTemp'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - loadCfg - Loading game config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - migrateCfgOption - File '/home/romeu/.config/steamtinkerlaunch/dxvk/230050.conf' does not exist - nothing to do
Sat Oct 21 03:34:59 PM WEST 2023 INFO - getRes - Detected screen resolution '3840x2160'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - pollWinRes - Using GEOM '--geometry=3816x2113+0+0' from '/home/romeu/.config/steamtinkerlaunch/guicfgs/3840x2160/230050/SteamTinkerLaunch-launchGameMenu.conf'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - openMenu - Starting createMenu "/dev/shm/steamtinkerlaunch/menutemp/gamemenu" "launchGameMenu" "/home/romeu/.config/steamtinkerlaunch/games/header/230050.jpg" "EMPTY" "EMPTY" "Settings for game 'DLC Quest (230050)'" "MainMenu" "MENU_GAME"
Sat Oct 21 03:34:59 PM WEST 2023 INFO - updateWinRes - Starting resolution-poll for '/home/romeu/.config/steamtinkerlaunch/guicfgs/3840x2160/230050/SteamTinkerLaunch-launchGameMenu.conf' with incoming '3816x2113'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - createMenu - got '9' elements: '0 /dev/shm/steamtinkerlaunch/menutemp/gamemenu launchGameMenu /home/romeu/.config/steamtinkerlaunch/games/header/230050.jpg EMPTY EMPTY Settings for game 'DLC Quest (230050)' MainMenu MENU_GAME'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - openMenu - Currently used tempfile is '/dev/shm/steamtinkerlaunch/menu.1oIwKM4x'
Sat Oct 21 03:34:59 PM WEST 2023 INFO - updateWinRes - Waiting for Window 'SteamTinkerLaunch-launchGameMenu'
Sat Oct 21 03:35:00 PM WEST 2023 INFO - updateWinRes - Window 'SteamTinkerLaunch-launchGameMenu' is running - polling the resolution
Sat Oct 21 03:35:01 PM WEST 2023 INFO - openMenu - Clicked '2' - 'MAIN MENU'
Sat Oct 21 03:35:01 PM WEST 2023 INFO - openMenu - exiting 'Favorite Menu' and opening 'Settings Menu'
Sat Oct 21 03:35:01 PM WEST 2023 INFO - MainMenu - Preparing to load Main Menu
Sat Oct 21 03:35:01 PM WEST 2023 SKIP - createDLSpekList - Skipping looking for available SpecialK versions, because USESPECIALK is disabled
Sat Oct 21 03:35:01 PM WEST 2023 INFO - updateWinRes - The Window 'SteamTinkerLaunch-launchGameMenu' was closed - the resolution didn't change - nothing to do
Sat Oct 21 03:35:01 PM WEST 2023 INFO - createDLReShadeList - Found the following 'ReShade' versions online '5.9.2!5.9.1!5.9.0'
Sat Oct 21 03:35:01 PM WEST 2023 INFO - prepareMenu - Opening Menu
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/global.conf' START
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/global.conf' STOP
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' START
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - Loading game config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf'
Sat Oct 21 03:35:01 PM WEST 2023 INFO - migrateCfgOption - File '/home/romeu/.config/steamtinkerlaunch/dxvk/230050.conf' does not exist - nothing to do
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' STOP
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/url.conf' START
Sat Oct 21 03:35:01 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/url.conf' STOP
Sat Oct 21 03:35:01 PM WEST 2023 SKIP - updateConfigFile - Config file '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' already at version 'v14.0.20231021-2'
Sat Oct 21 03:35:01 PM WEST 2023 INFO - linkGameCfg - Symlink '/home/romeu/.config/steamtinkerlaunch/gamecfgs/title/DLC Quest.conf' already exists
Sat Oct 21 03:35:01 PM WEST 2023 INFO - createWineList - Updating the Wine Dropdown List
Sat Oct 21 03:35:01 PM WEST 2023 INFO - getGameTextFiles - Searching for at least one editable textfile in '/run/media/romeu/GAMES/SteamLibrary/steamapps/compatdata/230050/pfx/drive_c/users/steamuser'
Sat Oct 21 03:35:02 PM WEST 2023 INFO - setShowPic - Determining game picture
Sat Oct 21 03:35:02 PM WEST 2023 INFO - setShowPic - Using '/home/romeu/.config/steamtinkerlaunch/games/header/230050.jpg' as game picture
Sat Oct 21 03:35:02 PM WEST 2023 INFO - setOPCustPath - Default path for custom exe file requester is '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:02 PM WEST 2023 INFO - getRes - Detected screen resolution '3840x2160'
Sat Oct 21 03:35:02 PM WEST 2023 INFO - pollWinRes - Using GEOM '--geometry=3816x2113+0+0' from '/home/romeu/.config/steamtinkerlaunch/guicfgs/3840x2160/230050/SteamTinkerLaunch-MainMenu.conf'
Sat Oct 21 03:35:02 PM WEST 2023 INFO - updateWinRes - Starting resolution-poll for '/home/romeu/.config/steamtinkerlaunch/guicfgs/3840x2160/230050/SteamTinkerLaunch-MainMenu.conf' with incoming '3816x2113'
Sat Oct 21 03:35:02 PM WEST 2023 INFO - updateWinRes - Waiting for Window 'SteamTinkerLaunch-MainMenu'
Sat Oct 21 03:35:02 PM WEST 2023 INFO - MainMenu - Game Info Gotten!
Sat Oct 21 03:35:03 PM WEST 2023 INFO - updateWinRes - Window 'SteamTinkerLaunch-MainMenu' is running - polling the resolution
Sat Oct 21 03:35:03 PM WEST 2023 INFO - MainMenu - Clicked '16' - 'PLAY'
Sat Oct 21 03:35:03 PM WEST 2023 INFO - MainMenu - exiting 'Settings Menu' and opening 'Game'
Sat Oct 21 03:35:03 PM WEST 2023 INFO - startSteamGame - Starting game
Sat Oct 21 03:35:03 PM WEST 2023 INFO - goBackToPrevFunction - 'MainMenu' closed
Sat Oct 21 03:35:03 PM WEST 2023 INFO - goBackToPrevFunction - 'MainMenu' closed
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - Second LoadCfg after menus: /home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf
Sat Oct 21 03:35:03 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' START
Sat Oct 21 03:35:03 PM WEST 2023 INFO - loadCfg - Loading game config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf'
Sat Oct 21 03:35:03 PM WEST 2023 INFO - migrateCfgOption - File '/home/romeu/.config/steamtinkerlaunch/dxvk/230050.conf' does not exist - nothing to do
Sat Oct 21 03:35:03 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/gamecfgs/id/230050.conf' STOP
Sat Oct 21 03:35:03 PM WEST 2023 INFO - setDxvkVars - Exporting variable 'DXVK_FRAME_RATE=none' to limit game framerate
Sat Oct 21 03:35:03 PM WEST 2023 INFO - autoCollectionSettings - Searching collections for game '230050' in '/home/romeu/.local/share/Steam/userdata/109895103/7/remote/sharedconfig.vdf'
Sat Oct 21 03:35:03 PM WEST 2023 INFO - autoCollectionSettings - Global Config '/usr/share/steamtinkerlaunch/collections/x64dbg.conf' found - loading its settings
Sat Oct 21 03:35:03 PM WEST 2023 INFO - loadCfg - '/usr/share/steamtinkerlaunch/collections/x64dbg.conf' START
Sat Oct 21 03:35:03 PM WEST 2023 INFO - loadCfg - '/usr/share/steamtinkerlaunch/collections/x64dbg.conf' STOP
Sat Oct 21 03:35:03 PM WEST 2023 SKIP - autoCollectionSettings - Collections Directory Config '' not found - skipping
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - sortGameArgs again, in case something changed above:
Sat Oct 21 03:35:03 PM WEST 2023 SKIP - sortGameArgs - Sorting the game arguments is disabled
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - symlinkSteamUser:
Sat Oct 21 03:35:03 PM WEST 2023 SKIP - symlinkSteamUser - Symlinking steamuser is disabled - Checking if a symlink needs to be reverted
Sat Oct 21 03:35:03 PM WEST 2023 SKIP - symlinkSteamUser - '/run/media/romeu/GAMES/SteamLibrary/steamapps/compatdata/230050/pfx/drive_c/users/steamuser' is no symlink - nothing to do
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - restoreSteamUser:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - restoreSteamUser - Using configured value 'none' for RESTORESTEAMUSER
Sat Oct 21 03:35:03 PM WEST 2023 INFO - restoreSteamUser - Restoration of steamuser data is disabled with RESTORESTEAMUSER being 'none
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - prepareProton:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - setRunWineServer - Initiated from 'setNewProtVars'
Sat Oct 21 03:35:03 PM WEST 2023 WARN - setRunWineServer - Couldn't find the wine binary for the proton in path '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton' - falling back to 'wine' for plain RUNWINE and 'wineserver' for RUNWINESERVER
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareProton - Checking the Proton version is not required, because Steam Launch Option is used. Proton version set in Steam is 'proton-unknown-946'
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - setLinGameVals :
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - CheckTweakLaunch:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - checkXliveless:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - chooseWinetricks:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - chooseWinetricks - Using systemwide winetricks with version '20230212'
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - checkExtDeps:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - checkExtDeps done
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - setWineVars:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - CheckWinetricksLaunch:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - CheckWineCfgLaunch:
Sat Oct 21 03:35:03 PM WEST 2023 WARN - checkWineCfgLaunch - Function was called but RUN_WINECFG ('0') was not enabled - skipping
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - CustomRegs:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - TogWindows:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - StateSteamWebHelper:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - InstallReshade:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - installDepth3DReshade:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - prepareLaunch - checkReshade:
Sat Oct 21 03:35:03 PM WEST 2023 INFO - autoCollectionSettings - Searching collections for game '230050' in '/home/romeu/.local/share/Steam/userdata/109895103/7/remote/sharedconfig.vdf'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - autoCollectionSettings - Global Config '/usr/share/steamtinkerlaunch/collections/x64dbg.conf' found - loading its settings
Sat Oct 21 03:35:04 PM WEST 2023 INFO - loadCfg - '/usr/share/steamtinkerlaunch/collections/x64dbg.conf' START
Sat Oct 21 03:35:04 PM WEST 2023 INFO - loadCfg - '/usr/share/steamtinkerlaunch/collections/x64dbg.conf' STOP
Sat Oct 21 03:35:04 PM WEST 2023 SKIP - autoCollectionSettings - Collections Directory Config '' not found - skipping
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setFullGameExePath - User did not enable 'CUSTOMCMDRESHADE' - Using the game's exe directory as the ReShade installation directory
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setFullGameExePath - Using SHADDESTDIR '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest' for 'SHADDESTDIR'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - useSpecialK:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setFullGameExePath - User did not enable 'CUSTOMCMDRESHADE' - Using the game's exe directory as the ReShade installation directory
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setFullGameExePath - Using SHADDESTDIR '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest' for 'SPEKDDIR'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - updateConfigEntry - Option 'SPEKDLLNAME' is updated with the new value 'auto' in config '230050.conf'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - configureGeoElf:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setFullGameExePath - User did not enable 'CUSTOMCMDRESHADE' - Using the game's exe directory as the ReShade installation directory
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setFullGameExePath - Using SHADDESTDIR '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest' for 'GEOELFDDIR'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - ChooseShaders:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - UseNyrnaz:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - UseReplay:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - CheckSBSVRLaunch:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkMO2:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - setNewRes:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - getRes - Detected screen resolution '3840x2160'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - customUserScriptStart:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - redirectSCDP:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - redirectSCDP - Using Proton version with minor version fix '8.0-4b' for compatdata directory name
Sat Oct 21 03:35:04 PM WEST 2023 INFO - redirectSCDP - Using regular compatdata '/run/media/romeu/GAMES/SteamLibrary/steamapps/compatdata/230050'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - updateWinRes - The Window 'SteamTinkerLaunch-MainMenu' was closed - the resolution didn't change - nothing to do
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkIGCSInjector:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkOpenVRFSR:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - CheckFWS:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - loadCustomVars - Empty global config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/customvars/global-custom-vars.conf' for user defined custom variables not loaded
Sat Oct 21 03:35:04 PM WEST 2023 INFO - loadCustomVars - Empty game-specific config '/home/romeu/.config/steamtinkerlaunch/gamecfgs/customvars/230050.conf' for user defined custom variables not loaded
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - CheckCustomLaunch:
Sat Oct 21 03:35:04 PM WEST 2023 SKIP - checkCustomLaunch - USECUSTOMCMD is '0' therefore skipping the custom program '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkUUUPatchLaunch:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - CheckStraceLaunch:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - CheckNetMonLaunch:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkWDIB:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkPulse:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - checkSteamAppIDFile:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - checkSteamAppIDFile - STEAMAPPIDFILE is disabled and there's no '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/steam_appid.txt' - nothing to do
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - setCompatDataTitle:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setCompatDataTitle - Symlink /home/romeu/.config/steamtinkerlaunch/compatdata/DLC Quest already exists
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setCompatDataTitle - Symlink '/home/romeu/.config/steamtinkerlaunch/compatdata/DLC Quest' already points to the correct directory '/run/media/romeu/GAMES/SteamLibrary/steamapps/compatdata/230050'
Sat Oct 21 03:35:04 PM WEST 2023 SKIP - delayGameForCustomLaunch - Nothing to do
Sat Oct 21 03:35:04 PM WEST 2023 INFO - prepareLaunch - launchSteamGame:
Sat Oct 21 03:35:04 PM WEST 2023 INFO - launchSteamGame - Initial game command is '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- /home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton waitforexitandrun /run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setSLRReap - SLRPROTONVER is not defined, this is fine as regular game launches don't pass this
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setSLRReap - Falling back to RUNPROTON which is '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setSLRReap - ## SLR is enabled via USESLR=1 - prepending SLR from command line to the current launch command
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setSLRReap - ## reaper command is enabled via USEREAP=1 - prepending to the current launch command
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setSLRReap - Adding SLR '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper --' to the launch command
Sat Oct 21 03:35:04 PM WEST 2023 INFO - gameArgs - Game doesn't use any command line arguments
Sat Oct 21 03:35:04 PM WEST 2023 INFO - launchSteamGame - Preparing proton command
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setProtonCmd - Not overriding Proton and using Proton provided by steam commandline '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - setProtonPathVersion - Using directly known 'proton-unknown-946' as Proton Version for '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - checkX64dbgLaunch - Starting 'x64dbg' for 'DLC (230050)'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - getArch - Architecture for '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe' is 32bit
Sat Oct 21 03:35:04 PM WEST 2023 INFO - checkX64dbgLaunch - Using 'x32dbg' as 'DLC' is 32bit
Sat Oct 21 03:35:04 PM WEST 2023 INFO - checkX64dbgLaunch - File '' does not exit - starting installer
Sat Oct 21 03:35:04 PM WEST 2023 INFO - getRes - Detected screen resolution '3840x2160'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - pollWinRes - Using GEOM '--geometry=1846x1282+0+0' from '/home/romeu/.config/steamtinkerlaunch/guicfgs/3840x2160/230050/SteamTinkerLaunch-DownloadX64DbgStatus.conf'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - StatusWindow - for 'Downloading 'x64dbg''
Sat Oct 21 03:35:04 PM WEST 2023 INFO - updateWinRes - Starting resolution-poll for '/home/romeu/.config/steamtinkerlaunch/guicfgs/3840x2160/230050/SteamTinkerLaunch-DownloadX64DbgStatus.conf' with incoming '1846x1282'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - updateWinRes - Waiting for Window 'SteamTinkerLaunch-DownloadX64DbgStatus'
Sat Oct 21 03:35:04 PM WEST 2023 INFO - notiShow - Message 'Downloading 'XXX'' should go to StatusWindow
Sat Oct 21 03:35:04 PM WEST 2023 INFO - dlCheck - Downloading ''
Sat Oct 21 03:35:04 PM WEST 2023 INFO - notiShow - Message 'Downloading '/home/romeu/.config/steamtinkerlaunch/downloads/x64dbg/'' should go to StatusWindow
Sat Oct 21 03:35:05 PM WEST 2023 INFO - dlCheck - 'wget -q --show-progress https://github.com/x64dbg/x64dbg/releases/download/snapshot/ -O /home/romeu/.config/steamtinkerlaunch/downloads/x64dbg/'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - notiShow - Message 'Download finished 'XXX'' should go to StatusWindow
Sat Oct 21 03:35:05 PM WEST 2023 INFO - notiShow - Message 'Extracting 'XXX'' should go to StatusWindow
Sat Oct 21 03:35:05 PM WEST 2023 INFO - dlX64Dbg - Download of '' to '/home/romeu/.config/steamtinkerlaunch/downloads/x64dbg/' was successful
Sat Oct 21 03:35:05 PM WEST 2023 INFO - notiShow - Message 'Done' should go to StatusWindow
Sat Oct 21 03:35:05 PM WEST 2023 SKIP - dlX64Dbg - Extracting of file '/home/romeu/.config/steamtinkerlaunch/downloads/x64dbg//' failed
Sat Oct 21 03:35:05 PM WEST 2023 INFO - checkX64dbgLaunch - Installing failed - can't start '/home/romeu/.config/steamtinkerlaunch/downloads/x64dbg/release/x32/x32dbg.exe' - skipping
Sat Oct 21 03:35:05 PM WEST 2023 INFO - launchSteamGame - Original incoming start command: '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- /home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton waitforexitandrun /run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - launchSteamGame - Final outgoing start command: '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- /home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton waitforexitandrun /run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/meta/id/general/230050.conf' START
Sat Oct 21 03:35:05 PM WEST 2023 INFO - loadCfg - '/home/romeu/.config/steamtinkerlaunch/meta/id/general/230050.conf' STOP
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - Getting the Game Window name:
Sat Oct 21 03:35:05 PM WEST 2023 SKIP - getGameWindowName - Already have the gamewindow name: 'DLC Quest' - skipping
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - Full start command is '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- /home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton waitforexitandrun /run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - Rebuilding STEAM_COMPAT_TOOL_PATHS variable:
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - Adding RUNPROTON '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - Adding '/home/romeu/.local/share/Steam/ubuntu12_32' because USESLR is enabled
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - Result: Set STEAM_COMPAT_TOOL_PATHS from '/run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental:/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper:/home/romeu/.local/share/Steam/steamapps/common/Steamworks Shared:/run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier' to '/home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental:/home/romeu/.local/share/Steam/ubuntu12_32'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - ## ORIGINAL INCOMING LAUNCH COMMAND: '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- /home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton waitforexitandrun /run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - ## STL LAUNCH COMMAND: '/home/romeu/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=230050 -- /home/romeu/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- /home/romeu/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/romeu/SSD_ONE/SteamLibrary/steamapps/common/Proton - Experimental/proton waitforexitandrun /run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe'
Sat Oct 21 03:35:05 PM WEST 2023 INFO - startGame - ## GAMESTART HERE ###
Sat Oct 21 03:35:05 PM WEST 2023 INFO - restoreOrgVars - Restoring previously cleared Variables
Sat Oct 21 03:35:05 PM WEST 2023 INFO - updateWinRes - Waiting for Window 'SteamTinkerLaunch-DownloadX64DbgStatus'
Sat Oct 21 03:35:06 PM WEST 2023 INFO - updateWinRes - Waiting for Window 'SteamTinkerLaunch-DownloadX64DbgStatus'
Sat Oct 21 03:35:07 PM WEST 2023 SKIP - updateWinRes - Timeout waiting for Window 'SteamTinkerLaunch-DownloadX64DbgStatus'
Sat Oct 21 03:35:25 PM WEST 2023 INFO - startGame - Symlink '/home/romeu/.config/steamtinkerlaunch/logs/gamelaunch/title/DLC Quest.log' already exists
Sat Oct 21 03:35:25 PM WEST 2023 INFO - startGame - Waiting for the process '87910' to finish
Sat Oct 21 03:35:25 PM WEST 2023 INFO - startGame - Process '87910' finished - closing
Sat Oct 21 03:35:25 PM WEST 2023 INFO - startGame - ## GAMESTOP after '20' seconds playtime
Sat Oct 21 03:35:25 PM WEST 2023 STOP - ######### SteamTinkerLaunch v14.0.20231021-2 #########
Sat Oct 21 03:35:25 PM WEST 2023 STOP - ######### CLEANUP #########
Sat Oct 21 03:35:25 PM WEST 2023 INFO - closeSTL - closing STL
Sat Oct 21 03:35:25 PM WEST 2023 INFO - updateConfigEntry - Emptying 'CUSTOMCMD' for '/home/romeu/.config/steamtinkerlaunch/default_template.conf'
Sat Oct 21 03:35:25 PM WEST 2023 INFO - updateConfigEntry - Option 'CUSTOMCMD' is updated with the new value 'echo' in config 'default_template.conf'
Sat Oct 21 03:35:25 PM WEST 2023 INFO - closeSTL - Creating '/dev/shm/steamtinkerlaunch/steamtinkerlaunch-closing.tmp'
Sat Oct 21 03:35:25 PM WEST 2023 INFO - checkPlayTime - The game was not started via 'steamtinkerlaunch' but directly from Steam - skipping Crash Requester
Sat Oct 21 03:35:25 PM WEST 2023 INFO - closeSTL - Game 'DLC Quest (230050)' exited - cleaning up custom processes if necessary
Sat Oct 21 03:35:25 PM WEST 2023 INFO - closeTrayIcon - Closing TrayIcon '79298'
Sat Oct 21 03:35:26 PM WEST 2023 INFO - closeSTL - Removing '/dev/shm/steamtinkerlaunch/steamtinkerlaunch-closing.tmp'
Sat Oct 21 03:35:26 PM WEST 2023 STOP - ######### closeSTL SteamTinkerLaunch v14.0.20231021-2 #########
Sat Oct 21 03:35:26 PM WEST 2023 INFO - writeLastRun - Recreating /home/romeu/.config/steamtinkerlaunch/logs/steamtinkerlaunch/lastrun.txt
Sat Oct 21 03:35:26 PM WEST 2023 INFO - storeMetaData - Saving metadata for game 'DLC Quest (230050)'
Sat Oct 21 03:35:26 PM WEST 2023 INFO - updateConfigEntry - 'GAMENAME' option missing in config '230050.conf' - adding a new line
Sat Oct 21 03:35:26 PM WEST 2023 INFO - getArch - Architecture for '/run/media/romeu/GAMES/SteamLibrary/steamapps/common/DLC Quest/DLC.exe' is 32bit
Sat Oct 21 03:35:26 PM WEST 2023 INFO - storeMetaData - '/home/romeu/.config/steamtinkerlaunch/meta/eval/id/evaluatorscript_230050.vdf' does not exists
```

**Comment by sonic2kk** [comment]
Interesting! I have never used this option since taking over the project, I'll investigate. Thanks for reporting!

Could you attach the log as a file instead of text please? It makes it easier to search through :-)

**Comment by abc123098123** [comment]
Sure thing!

[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/13061075/steamtinkerlaunch.log)

Hope this helps!

**Comment by sonic2kk** [comment]
One part of your log that stands out to me is:

```
Sat Oct 21 03:33:06 PM WEST 2023 INFO - setflatpak - started SteamTinkerLaunch from /usr/bin/steamtinkerlaunch
Sat Oct 21 03:33:06 PM WEST 2023 INFO - initAID - Set AID to PLACEHOLDERAID '31337'
```

Are you definitely launching SteamTinkerLaunch from Steam as a compatibility tool (DLC Quest is a Windows game as far as I can see)? I'm not sure it would make a huge amount of difference but it's just something I noticed.

**Comment by abc123098123** [comment]
> One part of your log that stands out to me is:
> 
> ```
> Sat Oct 21 03:33:06 PM WEST 2023 INFO - setflatpak - started SteamTinkerLaunch from /usr/bin/steamtinkerlaunch
> Sat Oct 21 03:33:06 PM WEST 2023 INFO - initAID - Set AID to PLACEHOLDERAID '31337'
> ```
> 
> Are you definitely launching SteamTinkerLaunch from Steam as a compatibility tool (DLC Quest is a Windows game as far as I can see)? I'm not sure it would make a huge amount of difference but it's just something I noticed.

Well, good point. I should clarify that: I am using proton as the compatibility tool but I am launching the game as `steamtinkerlaunch %command%` in the game steam launch options.

EDIT: Ok, I think I am doing one thing wrong here. Launching as command line options is intended for Native Linux games. I should try using steamtinkerlaunch as a compatibility tool instead.

**Comment by sonic2kk** [comment]
You need to use SteamTinkerLaunch as a compatibility tool for Proton games, the launch option is only for native Linux games, [as noted on the Readme](https://github.com/sonic2kk/steamtinkerlaunch#steam-compatibility-tool-proton-games)

I don't expect it to resolve much, but it is good practice :-)

**Comment by sonic2kk** [comment]
Please bear with me as well, I have not used this option before (I took over the project from the original creator who implemented this [2 years ago it looks like](https://github.com/sonic2kk/steamtinkerlaunch/wiki/X64dbg-Support)) but I'll do my best to investigate.

**Comment by abc123098123** [comment]
No worries, and I'm so sorry, I really was doing it the wrong way. I am going to try to use `steamtinkerlaunch` as a compatibility tool and I'll report back. Thanks!

**Comment by sonic2kk** [comment]
I can reproduce the error: `checkX64dbgLaunch - Installing failed - can't start '/home/user/.config/steamtinkerlaunch/downloads/x64dbg/release/x64/x64dbg.exe' - skipping`

I didn't test any 32bit games, but I imagine the result will be the same, except it'll complain about `x32dbg`.

**Comment by abc123098123** [comment]
So, using `steamtinkerlaunch` as a compatibility tool didn't fix it at all. It still does not download anything. The issue persists.

If I download it manually and put it in the correct path, `x64dbg` actually starts! But it will stay up without the game starting. If I close `x64dbg`, the game will then start. Shouldn't both programs start at the same time?

**Comment by sonic2kk** [comment]
So for the initial download failing, it seems like the way STL downloads this from GitHub no longer works. GitHub changed some things a while ago and we fixed it for some tools, but I guess others got forgotten, so I imagine this needs fixed first! :-)

> If I download it manually and put it in the correct path, x64dbg actually starts! But it will stay up without the game starting. If I close x64dbg, the game will then start. Shouldn't both programs start at the same time?

I would say they should both start. Does this mean the second part of your OP issue is fixed?

> I also tried to install x64dbg in the expected directory. After that, it does recognize that x64dbg exists and tries to run it, giving the following error

If that is fixed that is some good news, otherwise there are 3 things that will need to get sorted here:
1. Fix downloading x64dbg from GitHub. I would guess extraction still works and that it's only the downloading part that fails
2. Investigate why x64dbg doesn't start *with* the game, I have not looked at the code for how x64dbg is started yet so I'll need to look into it.
3. Investigate why x64dbg is sometimes not found

**Comment by abc123098123** [comment]
Indeed, I do confirm that using STL as a compatibility tool does fix the second part of the OP issue, fortunately!

**Comment by sonic2kk** [comment]
I think the error in `extWine64Run` then is caused by STL not being able to get some variables needed to start a Proton game when used as a launch option. Nice to hear that's one issue down!

I'll first take a look at getting x64dbg to download properly. STL previously used to wget the GitHub releases page and then used grep to get the part of the page with the releases download links. It can still get the page, but that page now dynamically loads releases, so running `wget -q https://github.com/x64dbg/x64dbg/releases/tag/snapshot -O test.html` (the commad STL uses to download the page) and opening it in a browser will show the releases page, but with that spinner for the assets section.

The fix for downloading x64dbg should hopefully just be a case of re-using some functions we have to download assets from GitHub.

**Comment by abc123098123** [comment]
Ok, great!

So, forgetting the issue about downloading `x64dbg`, let's imagine everything goes well and it installs to the expected directory: 

1. I define the game as `debug with x64dbg`, and I also use the `x64dbg` steam collection (just to be sure).
2. I launch the game normally, and x64dbg appears, as expected. But the game isn't running yet.
3. The game only starts *after* closing x64dbg.

Here is a STL log for this specific case:
[steamtinkerlaunch.log](https://github.com/sonic2kk/steamtinkerlaunch/files/13061128/steamtinkerlaunch.log)

**Comment by sonic2kk** [comment]
It looks like the game is maybe meant to be started by `x64dbg` tool, but I'm not sure yet. The log from that run is helpful, thanks for attaching it.

**Comment by abc123098123** [comment]
Ah, good point! I was thinking about both of them starting so I could use `x64dbg` to attach to the game. But indeed, I did try to start the game *through* `x64dbg`, and the game *did* start. So I guess that part really isn't a bug.

EDIT: but in that case, should it start the game after closing `x64dbg`, though?

**Comment by sonic2kk** [comment]
> EDIT: but in that case, should it start the game after closing x64dbg, though?

Yes, because once the start command for x64dbg ends, it moves onto the rest of the launch sequence. You can think of it as the Wine run command holds up moving through the rest of the script, preventing it from moving onto the part of launching the game.

If the desired behaviour is for x64dbg and the game to start at the same time, that can probably be achieved, but *is* this desired behaviour? I'll have to let you tell me, this is my first time using the program.

I should mention that manually extracting x64dbg did also get the program to open for me, so depending on the above, the main issue here is downloading x64dbg :-)

**Comment by abc123098123** [comment]
I agree that the main issue is downloading `x64dbg`.

About the startup of `x64dbg`, there are two ways of using it:

1. Run `x64dbg` and use it to initialize the intended program/game/whatever.
2. Run `x64dbg` and game at the same time, and attach to the game through `x64dbg` (this would be done manually by the user, but the wine prefix would have to be the same, AFAIK).

So, it all depends on the user, really. I prefer having both open and attach to the game through `x64dbg`. But that's just me, though.

EDIT: And maybe opening both at the same time would be better because of environment variables? I don't really know. Opening the game through `x64dbg` might not use the env vars correctly. But I don't really understand that part.

EDIT2: As an extension from the previous edit, if I try to open `Yakuza: Like a Dragon` through `x64dbg`, I get some game specific error. While opening it normally through Steam just works. This could be an issue with `x64dbg` itself, but the thing here is that I chose the `.exe` file that maybe runs the game. If the game was already running, I could just attach to it.

**Comment by sonic2kk** [comment]
Something interesting I noticed in the x64dbg log was this: 

```
Handling command line...
  "Z:\home\user\.config\steamtinkerlaunch\downloads\x64dbg\release\x64\x64dbg.exe" "wineserver: using server-side synchronization.
wine: Using setpriority to control niceness in the [-10,10] range
Z:\run\media\user\500GB SSD\Non-Steam Games\Touhou Fumo Racing\TouhouFumoRacing.exe"
File does not exist!
```

Aside from "File does not exist", the rest of this is the commandline output that STL passes over to x64dbg. It seems that sometime in the last almost 2 years since its implementation, the command that STL passes to x64dbg is wrong.

So STL *is* meant to give the game to x64dbg and I assume start it that way, but then once x64dbg is closed either game would close or even if it didn't, I think the game would open anyway.

> EDIT2: As an extension from the previous edit, if I try to open Yakuza: Like a Dragon through x64dbg, I get some game specific error. While opening it normally through Steam just works. This could be an issue with x64dbg itself, but the thing here is that I chose the .exe file that maybe runs the game. If the game was already running, I could just attach to it.

It sounds like a good solution would be to just run the programs separately and get the user to manually attach the process :-) But I'll investigate and see!

**Comment by abc123098123** [comment]
> So STL is meant to give the game to x64dbg and I assume start it that way, but then once x64dbg is closed either game would close or even if it didn't, I think the game would open anyway.

Makes sense. If I execute `x64dbg` like `./x64dbg.exe random-executable.exe`, `x64dbg` will open it, and I can say this because I tested a few seconds ago.

> It sounds like a good solution would be to just run the programs separately and get the user to manually attach the process :-) But I'll investigate and see!

I would definitely prefer that way. But I can only speak for myself. Maybe should you offer the user a choice, in this specific case?

**Comment by sonic2kk** [comment]
I have a work-in-progress fix locally for getting this program and the game to start together, and I can successfully attach the process.

Sorry for my ignorance here, but when I attach the game process to `x64dbg`, is it expected that the game will close when `x64dbg` closes?

**Comment by abc123098123** [comment]
>Sorry for my ignorance here, but when I attach the game process to x64dbg, is it expected that the game will close when x64dbg closes?

Yes, if you close `x64dbg` while it is attached to the game. If you detach the game from `x64dbg` and close `x64dbg`, the game should, probably, keep running.

**Comment by sonic2kk** [comment]
> Yes, if you close x64dbg while it is attached to the game. If you detach the game from x64dbg and close x64dbg, the game should, probably, keep running.

Can confirm I'm seeing this behaviour, thanks!

So locally I have two code paths that I can control by a comment. The first, I fixed the EXE path so it automatically getting attached to x64dbg. The game does not start on this path, it only gets attached.

The second path is where x64dbg and the game start together, but the process is not automatically attached. I need to do some code cleanup for this path as it's hooking into parts of the STL script that weren't meant for it, but I should be able to fix this up.

I think then we can add an option to toggle this on the GUI. Do you have any ideas on what would be the best way to do this? Maybe an option like "start x64dbg and game separately", and if so, which option should be the default? I'm asking for feedback because, well, this is a totally new area for me, so I'll trust the better judgement of someone who has used this software before :smile:

**Comment by abc123098123** [comment]
Thanks! This feature is great for people like me that do *not* want to install Windows, but still want to play around with Windows-only games, while checking its behavior in a debugger like `x64dbg`.

I do think an option is needed. Sometimes it is very useful to attach before actually running it because it's sometimes useful to understand how the initialization works. Therefore, the default option should be the first path: the game doesn't start but gets attached automatically on `x64dbg`. I think it's the more *sane* option.

**Comment by sonic2kk** [comment]
> I do think an option is needed. Sometimes it is very useful to attach before actually running it because it's sometimes useful to understand how the initialization works

This is a fantastic point, and now that you mention it I wholeheartedly agree. It makes sense to be a safe default.

It is also straightforward then to control whether the game launches when x64dbg is closed, because we can check if we disabled the option to attach the process x64dbg on startup.

Following this issue, the wiki will also be updated with some more information around this behaviour too.

> This feature is great for people like me that do not want to install Windows

The less reason to use Windows, the better in my opinion :-)

**Comment by abc123098123** [comment]
> It is also straightforward then to control whether the game launches when x64dbg is closed, because we can check if we disabled the option to attach the process x64dbg on startup.

Yes! I am also glad it helps you in that case.

> The less reason to use Windows, the better in my opinion :-)

Exactly, and you're helping people achieve that!

**Comment by sonic2kk** [comment]
Doing some tests and finalizing some wording, hoping to get a branch up with the changes to x64dbg execution. There is no fix for downloading x64dbg yet, it seems like that will be a little bit tricker.

Are you comfortable testing branches? If so, you'll be able to test out the new (hopefully working!) fixes for x64dbg's execution and see if it meets your needs. I'll send a reply once the branch is up.

**Comment by abc123098123** [comment]
> There is no fix for downloading x64dbg yet, it seems like that will be a little bit tricker.

Sure, take your time. There is a workaround and that's fine by me.

> Are you comfortable testing branches? If so, you'll be able to test out the new (hopefully working!) fixes for x64dbg's execution and see if it meets your needs. I'll send a reply once the branch is up.

I am, I'll test the branch when it's up.

**Comment by sonic2kk** [comment]
The branch is up at [`x64dbg-fix`](https://github.com/sonic2kk/steamtinkerlaunch/tree/x64dbg-fix). The version is bumped, so a nice visual way to see what you're on the right branch right away is that the version should be `v14.0.20231022-3 (x64dbg-fix)`.

This branch makes three changes:
- Fixes the game EXE path not being passed correctly to `x64dbg` -- It was getting confused with other Wine output, so we assume the last line of the `winepath` command will be the sanitised game path (which it should be in every valid case). This fixes the default path of having x64dbg launch with the game process attached, so you can view initialisation
- Fixes game process launching after closing `x64dbg`. Now if you launch it with the game process attached, when you close x64dbg, the game should be marked as closed from Steam, and the game process will not start up again when you close x64dbg.
- Adds a toggle to choose whether to attach the game process to x64dbg on startup or not. If enabled, x64dbg will start standalone with the game executable attached. If disabled, the game will start, and then x64dbg will start shortly after and you can manually attach the game process.

I use the term `x64dbg`, but I don't see any reason from the code point of view why this shouldn't work for 32bit programs as well, although I have not explicitly tested this yet.

Sadly, there is no fix yet for downloading x64dbg. I recognise that this is important so I am not brushing it off, however it is a less trivial fix. The way STL downloads programs from GitHub depends on their tags, and x64dbg does not use tags for their releases the same way other projects do. So I will need to investigate how to resolve this and properly download x64dbg. I will make a point of fixing it before closing this issue, so this is just an update to say I know it isn't fixed yet and that I am keeping it in mind.

My "workflow" so-to-speak is normally to get fixes up into a branch and then open a PR and merge. So because of that, if the fixes on the branch are working, I will merge that branch first, and then open a separate PR for the download fix. So these fixes will be merged in before the x64dbg downloading fix, and once they're merged, you should be able to use them in STL-git master without having to wait on me investigating the downloading issue, or stay on the branch for too long. If all goes well, I'm hoping to have the `x64dbg-fix` branch merged in before the weekend is out.

**Comment by abc123098123** [comment]
So, I tested the branch and it's working as expected, *both* paths!

**Comment by sonic2kk** [comment]
Awesome, I'm glad to hear it!

I'll do some cleanup, update the other langfiles, and hopefully get this merged tonight :-)

Thanks for the patience, this part of the code hadn't been touched in a long time and wasn't really a place I had ever spent any time, so even though it sucks you ran into it, now it'll be fixed for other users. Thanks for taking the time to report and test!

**Comment by abc123098123** [comment]
@sonic2kk 

Hey, thank *you* for actually taking the time to solve it. Very much appreciated, really!

**Comment by sonic2kk** [comment]
Sorry, I just realised I made a small mistake with how I was writing out to the config file. It shouldn't break anything but you may see some errors from the commandline along the lines of `steamtinkerlaunch: line 6859: export: `$DESC_X64DBG_ATTACHONSTARTUP=': not a valid identifier`.

I found the cause of it and I'll fix it before merging the PR, which is available at #950. Should be merged into master soon, then I'll take a look at how to fix downloading x64dbg :-)

**Comment by sonic2kk** [comment]
PR was merged into master, any STL version `v14.0.20231022-1` or above should have this fix.

Next up is figuring out the downloads issue. Once that's fixed, this issue can be closed :-)

**Comment by abc123098123** [comment]
Ok great! I guess that problem wasn't that important because I didn't have any issue. Thank you so much!

**Comment by sonic2kk** [comment]
No problem! Please let me know if you have any furrther problems and I'll try investigate asap :smile:

**Comment by sonic2kk** [comment]
Got a fix up for downloading x64dbg, at #952. Feel free to test if you have some time :-)

**Comment by abc123098123** [comment]
Great job. Just tested the branch [x64dbg-dl-fix](https://github.com/sonic2kk/steamtinkerlaunch/tree/x64dbg-dl-fix) and it works perfectly, seamlessly.

**Comment by sonic2kk** [comment]
Awesome! I'll get it merged. It's not the best fix, but I can revisit it later. The URLs it's using haven't changed since 2015, so it's reasonably safe, but not as robust as the checks we do for MO2/Vortex/HMM. Since the wget for x64dbg is only for that project though I think it's safe :-)

This should be available in master shortly!

**Comment by abc123098123** [comment]
Ok, sure! You may close this issue whenever you feel like it. Thank you, once again!

**Comment by sonic2kk** [comment]
Merged in, this issue was auto-closed by the merge.

Thanks for the report and glad we got this resolved quickly. It's always nice to see some of these lesser-known parts of STL get some fixes :smile:
